### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by Zumbalamambo** [google/automl#25]
How do I train on a custom dataset?

**Comment by chentairan** [comment]
I think you can convert your dataset to tf_record format

**Comment by mingxingtan** [comment]
https://github.com/google/automl/issues/34

**Comment by CraigWang1** [comment]
Has anyone successfully trained on a custom dataset?

I tried training on a custom dataset with one class and 1700 images in colab:
```
# train!
!python main.py --training_file_pattern=tfrecord/train* --model_dir=models --hparams="use_bfloat16=false,num_classes=1,learning_rate=0.001" --use_tpu=False \
  --model_name='efficientdet-d0' --train_batch_size 16 \
  --validation_file_pattern=tfrecord/val* --val_json_file='/content/unzipped_dset/bin_COCO/data/COCO/annotations/instances_val2017.json' \
  --mode='train_and_eval'
```
The loss looked normal after a few epochs:
![edet_0_loss](https://user-images.githubusercontent.com/55901251/77713100-c210a300-6f92-11ea-993c-6820cff68817.png)

However, when I did eval, I got 0 mAP:
```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
```
Has anyone else had this issue?

**Comment by honghande** [comment]
@CraigWang1 ，My customized dataset is labeled with lableme and converted into coco dataset. Do you need mask?
but
https://github.com/google/automl/issues/93
Do you know how to solve it？
